extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/methods/UserMethodsImplementor.java

```diff
@@ -14,26 +11,17 @@
 import io.quarkus.rest.data.panache.deployment.properties.ResourceProperties;
 
 /**
- * Propagate the user methods annotated with `@Transactional`.
- * This implementor is only used if Hibernate ORM is present.
+ * Propagate all the user methods. This is necessary when users use annotations with an interceptor binding
+ * like `@Transactional`.
  */
-public final class UserMethodsWithTransactionalImplementor implements MethodImplementor {
-
-    public static final DotName TRANSACTIONAL = DotName.createSimple("jakarta.transaction.Transactional");
-
-    private final Capabilities capabilities;
-
-    public UserMethodsWithTransactionalImplementor(Capabilities capabilities) {
-        this.capabilities = capabilities;
-    }
+public final class UserMethodsImplementor implements MethodImplementor {
 
     @Override
     public void implement(ClassCreator classCreator, ResourceMetadata resourceMetadata,
             ResourceProperties resourceProperties, FieldDescriptor resourceField) {
-        if (capabilities.isPresent(Capability.HIBERNATE_ORM) && resourceMetadata.getResourceInterface() != null) {
+        if (resourceMetadata.getResourceInterface() != null) {
             for (var methodInfo : resourceMetadata.getResourceInterface().methods()) {
-                // we only need to propagate the user methods annotated with `@Transactional`
-                if (methodInfo.hasAnnotation(TRANSACTIONAL)) {
+                if (methodInfo.isDefault()) {
```

Comment:
IIUIC, with this approach you are copying over all default methods and their annotations.
This definitely solves the bindings, I have no issue with that. 

However, it also carries over different annotations that would otherwise be ignored - such as `@Produces`. I am not sure that's something of a concern here, most likely not :)
Last but not least, you are very likely copying methods (most of them?) that you wouldn't even need to copy as they have either no annotation or no relevant annoation. A simple check for method having *any* annotation while being default might improve that?

Reply:
> A simple check for method having any annotation while being default might improve that?

this makes total sense. will update it.